apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/Options.kt

```diff
@@ -189,7 +189,14 @@ class Options(
      */
     val addJvmOverloads: Boolean = false,
     val addTypename: String = defaultAddTypename,
-    val requiresOptInAnnotation: String? = defaultRequiresOptInAnnotation
+    val requiresOptInAnnotation: String? = defaultRequiresOptInAnnotation,
+
+    /**
+     * Whether to merge fields in disjoint types without the FieldsInSetCanMerge validate.
+     *
+     * If true, disable the FieldsInSetCanMerge validation.
```

Comment:
I think the meaning of the flag has been inverted: 
- `fieldsOnDisjointTypesMustMerge = true` means the validation is run (the default)
- `fieldsOnDisjointTypesMustMerge = false` means the validation is disabled

```suggestion
     * Whether fields with different shape are disallowed to be merged in disjoint types.
     * 
     * Note: setting this to `false` relaxes the standard GraphQL [FieldsInSetCanMerge](https://spec.graphql.org/draft/#FieldsInSetCanMerge()) validation which may still be 
     * run on the backend.
     *
     * See also [issue 4320](https://github.com/apollographql/apollo-kotlin/issues/4320)
     *
     * Default: true
```

(Also should it be `fieldsOnDisjointTypesMustMerge`? I think plural makes more sense.)

Reply:
Whoops, sorry, the comment is wrong. I will fix these.